test/refunder-test.ts

```diff
@@ -0,0 +1,295 @@
+			const value = ethToWei('2');
+			const res = await owner.sendTransaction({
+				value: value,
+				to: refunder.address
+			});
+	
+			await res.wait();
+
+			await expect(refunder.connect(addr1).withdraw(value)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+
+		it("Not Owner should NOT withdraw ETHs from Refunder if balance is 0", async function() {
+		
+			const value = ethToWei('2');
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+
+			expect('0').to.be.eq(balanceOfRefunderBefore.toString());
+			await expect(refunder.connect(addr1).withdraw(value)).to.be.revertedWith(NOT_AN_OWNER);
+
+		});
+	});
+
+	describe('Max Gas Price', () => {
+		it("Owner should get max gas price", async function() {
+			const gasPrice = await refunder.maxGasPrice();
+			
+			expect('0').to.be.eq(gasPrice.toString());
+		});
+
+		it("Owner should set max gas price", async function() {
+			const gasPrice = await refunder.maxGasPrice();
+
+			const value = 152;
+
+			let res = await refunder.setMaxGasPrice(value);
+			await res.wait();
+
+			const updatedGasPrice = await refunder.maxGasPrice();
+
+			expect(updatedGasPrice.toString()).to.be.eq(value.toString());
+
+		});
+	
+		it("Not Owner should get max gas price", async function() {
+			const gasPrice = await refunder.connect(addr1).maxGasPrice();
+			
+			expect('0').to.be.eq(gasPrice.toString());
+		});
+
+		it("Not Owner should NOT set max gas price", async function() {
+			await expect(refunder.connect(addr1).setMaxGasPrice(191)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+	});
+
+	describe('Whitelist Refundable', () => {
+
+		it("Owner should be able to check for refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			// address, bytes4
+			const res = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			
+			expect(res).to.be.eq(false);
+		});
+
+		it("Owner should be able to add refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			// address, bytes4
+			const resBefore = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resBefore).to.be.eq(false);
+
+			const res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, true);
+			await res.wait();
+
+			const resAfter = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfter).to.be.eq(true);
+		});
+
+		it("Owner should be able to edit refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
```

Comment:
Maybe you could extract this to a function that you call several times. F.e `prepareRefundable`